src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -350,26 +302,31 @@ public IEnumerable<Detection> DetectSecrets(string input)
 
                 foreach (var detection in regexSecret.GetDetections(input, _generateCorrelatingIds, DefaultRegexRedactionToken, _regexEngine))
                 {
-                    yield return detection;
+                    if (detection.Length >= MinimumSecretLength)
+                    {
+                        detections.Add(detection);
+                    }
                 }
             }
 
-            foreach (SecretLiteral secretLiteral in EncodedSecretLiterals)
+            foreach (SecretLiteral secretLiteral in _encodedSecretLiterals)
             {
+                if (secretLiteral.Value.Length < MinimumSecretLength)
```

Comment:
>MinimumSecretLength [](http://example.com/codeflow?start=49&length=19)

Every regex should be enforcing minimal lengths not this API. Sorry for raising this point again.

Reply:
Yes, this minimum length feature sucks, see other thread, lol. 

But as long as we have both of these:

1. Public API that accepts arbitrary regex.
2. Public API that accepts arbitrary minimum length.

I think it follows that we have should have this check.